File: Demos/HiToday/app/src/main/java/top/iqqcode/hitoday/base/BaseActivity.java
package top.iqqcode.hitoday.base;

import androidx.appcompat.app.AppCompatActivity;

import org.xutils.common.Callback;
import org.xutils.http.RequestParams;
import org.xutils.x;

/**
 * @Author: iqqcode
 * @Date: 2020-12-16 22:03
 * @Description:获取网络数据
 */
public class BaseActivity extends AppCompatActivity implements Callback.CommonCallback<String> {
    public void loadData(String url) {
        RequestParams requestParams = new RequestParams(url);
        // 当前类继承CommonCallback接口，当前对象属于CommonCallback的对象(向上转型)
        x.http().get(requestParams, this);
    }

    @Override
    public void onSuccess(String result) {

    }

    @Override
    public void onError(Throwable ex, boolean isOnCallback) {

    }

    @Override
    public void onCancelled(CancelledException cex) {

    }

    @Override
    public void onFinished() {

    }
}

File: Android-Core/01-AndroidCore/d-BroadcastReceiver/01Broadcast/forceout/src/main/java/top/iqqcode/forceout/BaseActivity.kt
package top.iqqcode.forceout

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity


/**
 * @Author: iqqcode
 * @Date: 2022-02-13 11:15
 * @Description:定义父类BaseActivity
 */
open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        // 创建活动时，将其加入管理器中
        ActivityCollector.addActivity(this)
    }

    override fun onDestroy() {
        super.onDestroy()
        // 销毁活动时，将其从管理器中移除
        ActivityCollector.removeActivity(this)
    }
}